src/lifecycle/attribute.js

```diff
@@ -1,25 +1,10 @@
-import data from '../util/data';
-
 export default function (opts) {
   let callback = opts.attribute || function () {};
-
   return function (name, oldValue, newValue) {
-    let info = data(this);
-    let attributeToPropertyMap = info.attributeToPropertyMap || {};
-
     callback(this, {
       name: name,
-      newValue: newValue,
-      oldValue: oldValue
+      newValue: newValue === null ? undefined : newValue,
+      oldValue: oldValue === null ? undefined : oldValue
```

Comment:
Yeah, but moreover it's for consistency across the board. We return `undefined` for empty values instead of null.